test/e2e/common/downward_api.go

```diff
@@ -104,6 +104,56 @@ var _ = ginkgo.Describe("[sig-node] Downward API", func() {
 		testDownwardAPI(f, podName, env, expectations)
 	})
 
+	ginkgo.It("should provide host IP and pod IP as an env var if pod uses host network [LinuxOnly]", func() {
+		podName := "downward-api-" + string(uuid.NewUUID())
+		env := []v1.EnvVar{
+			{
+				Name: "HOST_IP",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "status.hostIP",
+					},
+				},
+			},
+			{
+				Name: "POD_IP",
+				ValueFrom: &v1.EnvVarSource{
+					FieldRef: &v1.ObjectFieldSelector{
+						APIVersion: "v1",
+						FieldPath:  "status.podIP",
+					},
+				},
+			},
+		}
+
+		expectations := []string{
+			fmt.Sprintf("OK"),
+		}
+
+		pod := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   podName,
+				Labels: map[string]string{"name": podName},
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{
+					{
+						Name:    "dapi-container",
+						Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+						Command: []string{"sh", "-c", `[[ "${HOST_IP:?}" == "${POD_IP:?}" ]] && echo 'OK' || echo`},
```

Comment:
maybe I'm misreading this, but I expect `echo` will just output an empty string instead of the env vars, and it's not clear to me that `MatchContainerOutput` (which ultimately gets called by `testDownwardAPIUsingPod`) would output the env vars for debugging purposes

Reply:
yes, I messed up the github suggestion box earlier, intention was:

```suggestion
						Command: []string{"sh", "-c", `[[ "${HOST_IP:?}" == "${POD_IP:?}" ]] && echo 'OK' || echo "HOST_IP: '${HOST_IP}' != POD_IP: '${POD_IP}'"`},
```

Reply:
Tested, was my fault, I should verify before committing 😄 , you can see now the output

```
  Unexpected error:
      <*errors.errorString | 0xc001dc3ad0>: {
          s: "expected \"OK\" in container output: Expected\n    <string>: HOST_IP: '2001:db8:1::242:ac11:3' != POD_IP: '2001:db8:1::242:ac11:3'\n    \nto match regular expression\n    <string>: OK",
      }
      expected "OK" in container output: Expected
          <string>: HOST_IP: '2001:db8:1::242:ac11:3' != POD_IP: '2001:db8:1::242:ac11:3'

      to match regular expression
          <string>: OK
  occurred
```